FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -29,14 +29,91 @@ public final class CoreDataFeedStore: FeedStore {
 	}
 
 	public func retrieve(completion: @escaping RetrievalCompletion) {
-		fatalError("Must be implemented")
+		let context = self.context
+		context.perform {
+			do {
+				let fetchedFeedRequest: NSFetchRequest<CoreDataFeed> = CoreDataFeed.fetchRequest()
+				let result = try context.fetch(fetchedFeedRequest)
+
+				guard let coreDataFeed = result.first,
+				      let timestamp = coreDataFeed.timestamp,
+				      let coreDataFeedArray = (coreDataFeed.coreDataFeedImages?.array as? [CoreDataFeedImage]) else {
+					completion(.empty)
+					return
+				}
+
+				completion(.found(feed: coreDataFeedArray.toSortedLocal(), timestamp: timestamp))
+			} catch {
+				completion(.failure(error))
+			}
+		}
 	}
 
 	public func insert(_ feed: [LocalFeedImage], timestamp: Date, completion: @escaping InsertionCompletion) {
-		fatalError("Must be implemented")
+		let context = self.context
+		context.performAndWait { [weak self] in
```

Comment:
There's no need to weakify `self` in this context. Remove `[weak self]`.

Moreover, `performAndWait` will block clients. Use `perform` instead.

```suggestion
		context.perform {
```

Reply:
@caiozullo, I have question. in CodableFeedStore realization, you used custom DispatchQueue, and on deletion and insertation methods custom queue was used as serial
My questions: Context manages updates on it's own queue serialy?

Reply:
@alexeybezrodniy, yes, the context's queue is serial. 

We have tests to check this behavior as well - if the tests pass, it's all good ✅